numpy/ma/core.py

```diff
@@ -6109,8 +6110,26 @@ def __reduce__(self):
         """
         return (self.__class__, ())
 
-
 masked = masked_singleton = MaskedConstant()
+
+def _return_masked(out):
+    """Return the value "masked", writing it to out if out is not None"""
+    if out is not None:
+        if out.size != 1:
+            raise Exception("out has wrong shape")
+
+        if isinstance(out, MaskedArray):
+            out.flat = 0
+            out.__setmask__(True)
+        else:
+            if out.dtype.kind in 'biu':
+                errmsg = "Masked data information would be lost in one or "\
+                         "more location."
+                raise MaskError(errmsg)
+            out.flat = np.nan
```

Comment:
I assume this use `np.nan` logic was there before? Seems fine to me. Not sure about the "biu" check, maybe the other way around? Might check explicitly for inexact, datetime/timedelta or object as well. Or just add string/unicode which is probably right.